src/callbacks.cc

```diff
@@ -458,10 +458,13 @@ void async_cb_handler(uv_async_t *handle)
 {
   NotifInfo *notif;
 
-  mutex::scoped_lock sl(zqueue_mutex);
-
-  while (!zqueue.empty()) {
-    notif = zqueue.front();
+  while (true) {
+    {
+      mutex::scoped_lock sl(zqueue_mutex);
+      if (zqueue.empty()) break;
+      notif = zqueue.front();
+      zqueue.pop();
```

Comment:
The main problem I see is that the pop() method [will call the notification destructor](http://www.cplusplus.com/reference/queue/queue/pop/) - thus ww'll probably need to clone it...

Reply:
`zqueue` is a queue of pointers, not a queue of objects, so the destructor will not get called here.